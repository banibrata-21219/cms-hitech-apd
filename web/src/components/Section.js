import PropTypes from 'prop-types';
import React from 'react';

import Collapsible from './Collapsible';
import Instruction from './Instruction';
import { t } from '../i18n';

const SectionTitle = ({ children }) => (
  <h2 className="mt1 mb2 h1 sm-h0 teal light">{children}</h2>
);

SectionTitle.propTypes = {
  children: PropTypes.node.isRequired
};

const SectionDesc = ({ children }) => <p className="mb4 text-xl">{children}</p>;

SectionDesc.propTypes = {
  children: PropTypes.node.isRequired
};

const Section = ({ children, id, resource }) => {
  const title = t([resource, 'title'], { defaultValue: false });
  const subheader = t([resource, 'subheader'], { defaultValue: false });
  const helptext = t([resource, 'helpText'], { defaultValue: false });

  return (
    <section id={id} className="py2 border-bottom border-grey border-width-3">
      {title && <SectionTitle>{title}</SectionTitle>}
      {subheader && <div>{subheader}</div>}
      {helptext && <SectionDesc>{helptext}</SectionDesc>}
      {children}
    </section>
  );
};

Section.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  resource: PropTypes.string
};

Section.defaultProps = {
  id: null,
  resource: null
};

const Subsection = ({ children, id, nested, open, resource }) => {
  const title = t([resource, 'title'], { defaultValue: '' });

  return (
    <Collapsible id={id} title={title} open={open} nested={nested}>
      <Instruction source={`${resource}.instruction`} />
      {children}
    </Collapsible>
  );
};

Subsection.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  nested: PropTypes.bool,
  open: PropTypes.bool,
  resource: PropTypes.string
};

Subsection.defaultProps = {
  resource: null,
  id: null,
  nested: false,
  open: false
};

export default Section;
export { Section, SectionDesc, SectionTitle, Subsection };
