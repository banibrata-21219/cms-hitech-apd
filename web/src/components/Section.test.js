import { shallow } from 'enzyme';
import React from 'react';

import { Section, SectionDesc, SectionTitle, Subsection } from './Section';

describe('Section component', () => {
  test('renders correctly with no resource', () => {
    const component = shallow(<Section>test child</Section>);
    expect(component).toMatchSnapshot();
  });

  test('renders correctly with resource containing title, subheader, and helptext', () => {
    const component = shallow(
      <Section resource="activities.goals">test child</Section>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('SectionDesc component', () => {
  test('renders correctly', () => {
    const component = shallow(<SectionDesc>test child</SectionDesc>);
    expect(component).toMatchSnapshot();
  });
});

describe('SectionTitle component', () => {
  test('renders correctly', () => {
    const component = shallow(
      <SectionTitle>
        <div className="test child" />
      </SectionTitle>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Subsection component', () => {
  test('renders correctly', () => {
    const component = shallow(
      <Subsection resource="activities.goals">test child</Subsection>
    );
    expect(component).toMatchSnapshot();
  });
});
