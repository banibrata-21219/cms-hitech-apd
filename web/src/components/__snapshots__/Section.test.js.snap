// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders correctly with no resource 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    id={null}
    resource={null}
  >
    test child
  </Section>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        false,
        false,
        "test child",
      ],
      "className": "py2 border-bottom border-grey border-width-3",
      "id": null,
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      false,
      "test child",
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          false,
          "test child",
        ],
        "className": "py2 border-bottom border-grey border-width-3",
        "id": null,
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
        "test child",
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Section component renders correctly with resource containing title, subheader, and helptext 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    id={null}
    resource="activities.goals"
  >
    test child
  </Section>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SectionTitle>
          Performance Goals and Benchmarks
        </SectionTitle>,
        false,
        false,
        "test child",
      ],
      "className": "py2 border-bottom border-grey border-width-3",
      "id": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Performance Goals and Benchmarks",
        },
        "ref": null,
        "rendered": "Performance Goals and Benchmarks",
        "type": [Function],
      },
      false,
      false,
      "test child",
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SectionTitle>
            Performance Goals and Benchmarks
          </SectionTitle>,
          false,
          false,
          "test child",
        ],
        "className": "py2 border-bottom border-grey border-width-3",
        "id": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Performance Goals and Benchmarks",
          },
          "ref": null,
          "rendered": "Performance Goals and Benchmarks",
          "type": [Function],
        },
        false,
        false,
        "test child",
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`SectionDesc component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionDesc>
    test child
  </SectionDesc>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test child",
      "className": "mb4 text-xl",
    },
    "ref": null,
    "rendered": "test child",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test child",
        "className": "mb4 text-xl",
      },
      "ref": null,
      "rendered": "test child",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`SectionTitle component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionTitle>
    <div
      className="test child"
    />
  </SectionTitle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="test child"
      />,
      "className": "mt1 mb2 h1 sm-h0 teal light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "test child",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="test child"
        />,
        "className": "mt1 mb2 h1 sm-h0 teal light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "test child",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Subsection component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Subsection
    id={null}
    nested={false}
    open={false}
    resource="activities.goals"
  >
    test child
  </Subsection>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bgColor": "white",
      "btnBgColor": "white",
      "btnColor": "blue",
      "children": Array [
        <Instruction
          args={null}
          reverse={false}
          source="activities.goals.instruction"
        />,
        "test child",
      ],
      "id": null,
      "nested": false,
      "onChange": null,
      "open": false,
      "sticky": true,
      "title": "Performance Goals and Benchmarks",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "args": null,
          "reverse": false,
          "source": "activities.goals.instruction",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "test child",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bgColor": "white",
        "btnBgColor": "white",
        "btnColor": "blue",
        "children": Array [
          <Instruction
            args={null}
            reverse={false}
            source="activities.goals.instruction"
          />,
          "test child",
        ],
        "id": null,
        "nested": false,
        "onChange": null,
        "open": false,
        "sticky": true,
        "title": "Performance Goals and Benchmarks",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "args": null,
            "reverse": false,
            "source": "activities.goals.instruction",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "test child",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
